### PR TITLE
Eilen/omniauth ets

### PR DESCRIPTION
**Pull request by eilenshahbaz**
Per ETS:
```Hi Eilen,

Thanks for registering your URL pattern(s). These are now authorized for the CAS Servers auth-test.b.e and auth.b.e as noted below.

CAS authZ is enabled for your service as follows:

Students, and members of the edu:berkeley:app:auth-cas:communities_at_cal:communities_at_cal-allow
```
I'm assuming auth-test.b.e and auth.b.e refer to the servers specified here: https://calnetweb.berkeley.edu/calnet-technologists/cas/casifying-your-web-application-or-web-server/first-level-cas-service-urls

Unable to play with it locally since these servers don't recognize localhost as a registered client/provider.

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -19,14 +21,10 @@ def signin
     redirect_to "/auth/cas"
   end
 
-  def signout
```
Comment (kaandogusoy) [inline app/controllers/sessions_controller.rb]: not sure why you changed the name, does it change the way it works

File: spec/controllers/sessions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
```
Comment (kaandogusoy) [inline spec/controllers/sessions_controller_spec.rb:1]: looks good I might add another test for sad path later tonight

File: app/views/placeholder/landing.html.erb [hunk 3]
```diff
@@ -34,8 +34,11 @@
           <%=image_tag("LEAD logo.png", :align=> "middle", :style => "width:300px")%>
           <h1>Communities@Cal</h1>
           <p>Please login through CalNet to continue.</p> 
-          <%= form_tag login_path, :method => 'post' do %>
-            <input id="login-button" type="submit" value="Placeholder Login" class="btn-primary"/>
+          <%= form_tag signin_path, :method => 'get' do %>
+          <div class="col-md-4 offset-md-4">
+            <input id="login-button" type="submit" value="Sign in" class="btn btn-primary btn-block"
+                   style="background: linear-gradient(#2683bc, #1f7099), #1f7099"/>
```
Comment (melvinhernandez) [inline app/views/placeholder/landing.html.erb:40]: Ideally we should not do inline style to help with maintainability, but not a blocker for this feature!

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    reset_session
+    session[:user_id] = @user.id
+    session[:ticket] = auth[:credentials][:ticket]
+    session[:auth] = auth
+    session[:logged_in] = true
+    redirect_to root_url
+  end
+
+  def failure
+    session[:logged_in] = false
```
Comment (melvinhernandez) [inline app/controllers/sessions_controller.rb:16]: Looks like this method is not being tested, so just add one more test for this to get it to 100% 👍

**Review by melvinhernandez — CHANGES_REQUESTED** [review]
The authentication and tests look pretty good! 🤙
Only blockers before merging are:
- [x] Add test for `failure` method in `sessions_controller.rb` to get it to 100%.
- [x] Resolve merge conflicts with master branch for `Gemfile` and `schema.rb`.

**Review by melvinhernandez — APPROVED** [review]

**Pull request merged**